Source/Core/Common/Arm64Emitter.h

```diff
@@ -903,14 +927,14 @@ class ARM64XEmitter
   }
   void ORRI2R(ARM64Reg Rd, ARM64Reg Rn, u64 imm, ARM64Reg scratch = ARM64Reg::INVALID_REG);
   void EORI2R(ARM64Reg Rd, ARM64Reg Rn, u64 imm, ARM64Reg scratch = ARM64Reg::INVALID_REG);
-  void CMPI2R(ARM64Reg Rn, u64 imm, ARM64Reg scratch = ARM64Reg::INVALID_REG);
 
   void ADDI2R_internal(ARM64Reg Rd, ARM64Reg Rn, u64 imm, bool negative, bool flags,
                        ARM64Reg scratch);
   void ADDI2R(ARM64Reg Rd, ARM64Reg Rn, u64 imm, ARM64Reg scratch = ARM64Reg::INVALID_REG);
   void ADDSI2R(ARM64Reg Rd, ARM64Reg Rn, u64 imm, ARM64Reg scratch = ARM64Reg::INVALID_REG);
   void SUBI2R(ARM64Reg Rd, ARM64Reg Rn, u64 imm, ARM64Reg scratch = ARM64Reg::INVALID_REG);
   void SUBSI2R(ARM64Reg Rd, ARM64Reg Rn, u64 imm, ARM64Reg scratch = ARM64Reg::INVALID_REG);
+  void CMPI2R(ARM64Reg Rn, u64 imm, ARM64Reg scratch = ARM64Reg::INVALID_REG);
```

Comment:
What do you think about dropping the default value of `ARM64Reg::INVALID_REG`? If no register is specified, the caller should not use the I2R method.

Reply:
There are actually some cases where the value is guaranteed to not need a scratch register even though the value isn't known at compile time, like here: https://github.com/dolphin-emu/dolphin/blob/cfcc994f6c6966172c108ee73347d7ad3f0ae717/Source/Core/Core/PowerPC/JitArm64/JitArm64_SystemRegisters.cpp#L717

Reply:
Ah, but you mean it would be possible to just call `EncodeLogicalImm` at runtime... Yeah, I suppose that makes sense. I'll get rid of the default argument for the bitwise instructions. (But not for `CMPI2R`, because that one is not a bitwise instruction :))